server/utils.go

```diff
@@ -149,8 +149,8 @@ func Base64DecodePaddingAgnostic(s string) ([]byte, error) {
 // RemoveDuplicatesFromSlice returns a slice with all the duplicates removed from the input slice.
 func RemoveDuplicatesFromSlice[T comparable](slice []T) []T {
 	// We are using the allKeys map as a set here
-	allKeys := make(map[T]struct{})
-	var list []T
+	allKeys := make(map[T]struct{}, len(slice))
+	list := make([]T, 0, len(slice))
```

Comment:
if there are a lot of duplicates, we may be over allocating memory here

Reply:
but that would be in cases of large slices with many duplicates, otherwise this is probably ok

Reply:
The recent Go performance recommendation I've heard is that it is better to overallocate than underallocate. We don't want to keep allocating new memory for new slices due to append operation.